src/main/java/jp/ac/keio/bio/fun/xitosbml/cli/CliRun.java

```diff
@@ -10,16 +11,67 @@
 
 public class CliRun implements Callable<Integer> {
 
-	@Option(names = "-i", required = true, description = "The path to input image file")
+	@Option(names = "-i", required = true, description = "The path to input image file or folder containing images")
 	String inputValue;
-	@Option(names = "-o", required = true, description = "The path to output XML file")
+	@Option(names = "-o", required = true, description = "The path to output XML file if input is image file")
 	String outputValue;
 
+	// Function to check whether path specified is a folder or not
+	public void listFilesForFolder(final File folder) {
+
+		// Instantiating CliMainImgSpatial class
+		CliMainImgSpatial cliMain = new CliMainImgSpatial();
+
+		if (!folder.isDirectory()) {
+			// The input is path to an image file and hence can be processed directly
+			cliMain.runCli(inputValue, outputValue);
+		} else {
+			// The input is path to a folder containing image files
+			for (final File fileEntry : folder.listFiles()) {
+				if (fileEntry.isDirectory()) {
+					listFilesForFolder(fileEntry);
+				} else {
+					if (fileEntry.isFile()) {
+						String temp = folder.getAbsolutePath() + File.separator + fileEntry.getName();
+						// System.out.println(temp);
+
+						// Checking if file is an image
+						if (((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase()).equals("tif"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("tiff"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("bmp"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("dcm"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("fits"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("pdm"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("gif"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("jpeg"))
+								|| ((temp.substring(temp.lastIndexOf('.') + 1, temp.length()).toLowerCase())
+										.equals("png"))) {
```

Comment:
Maybe create a new variable to store the extension? So this string manipulation is done at most once.